### PR TITLE
Add playback of background/title music in game list

### PR DESCRIPTION
**Pull request by tGecko**
Upon selecting a game from game list/grid view, plays it's title music (`CUSAXXXXXX/sce_sys/snd0.at9`) if available.
Includes a setting to turn it on or off.

Special thanks to @cselip !

### CODE REVIEW

File: src/qt_gui/background_music_player.cpp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+BackgroundMusicPlayer::BackgroundMusicPlayer(QObject* parent) : QObject(parent) {
+    m_mediaPlayer = new QMediaPlayer(this);
+    m_audioOutput = new QAudioOutput(this);
+    m_mediaPlayer->setAudioOutput(m_audioOutput);
+}
+
+BackgroundMusicPlayer::~BackgroundMusicPlayer() {
+    delete m_mediaPlayer;
+    delete m_audioOutput;
```
Comment (roamic) [inline src/qt_gui/background_music_player.cpp]: AFAIK you should not explicitly delete objects that are part of Qt's object model (created with `parent` object parameter) as they will be deleted when `parent` is destroyed.

File: src/qt_gui/main_window.cpp [hunk 2]
```diff
@@ -484,9 +484,30 @@ void MainWindow::CreateConnects() {
...
+    if (isGameRunning || !Config::getPlayBGM()) {
+        BackgroundMusicPlayer::getInstance().stopMusic();
+        return;
+    }
+    QString snd0path = "";
+    int itemID = isTableList ? m_game_list_frame->currentItem()->row()
+                             : m_game_grid_frame->crtRow * m_game_grid_frame->columnCnt +
+                                   m_game_grid_frame->crtColumn;
+
+    snd0path = QString::fromStdString(m_game_info->m_games[itemID].snd0_path);
```
Comment (roamic) [inline src/qt_gui/main_window.cpp]:
```suggestion
    const auto snd0path = QString::fromStdString(m_game_info->m_games[itemID].snd0_path);
```

File: src/qt_gui/background_music_player.cpp [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+void BackgroundMusicPlayer::playMusic(const QString& snd0path) {
+    if (snd0path.isEmpty()) {
+        stopMusic();
+        return;
+    }
+    QUrl newMusic = QUrl::fromLocalFile(snd0path);
```
Comment (roamic) [inline src/qt_gui/background_music_player.cpp]:
```suggestion
    const QUrl newMusic = QUrl::fromLocalFile(snd0path);
```

File: src/qt_gui/game_grid_frame.cpp [hunk 4]
```diff
@@ -39,6 +39,15 @@ GameGridFrame::GameGridFrame(std::shared_ptr<GameInfoClass> game_info_get, QWidg
     });
 }
 
+void GameGridFrame::PlayBackgroundMusic(QTableWidgetItem* item) {
+    if (!item) {
+        BackgroundMusicPlayer::getInstance().stopMusic();
+        return;
+    }
+    QString snd0path = QString::fromStdString(m_game_info->m_games[item->row()].snd0_path);
```
Comment (roamic) [inline src/qt_gui/game_grid_frame.cpp]:
```suggestion
    const QString snd0path = QString::fromStdString(m_game_info->m_games[item->row()].snd0_path);
```

File: src/qt_gui/game_list_frame.cpp [hunk 5]
```diff
@@ -68,6 +68,15 @@ GameListFrame::GameListFrame(std::shared_ptr<GameInfoClass> game_info_get, QWidg
     });
 }
 
+void GameListFrame::PlayBackgroundMusic(QTableWidgetItem* item) {
+    if (!item) {
+        BackgroundMusicPlayer::getInstance().stopMusic();
+        return;
+    }
+    QString snd0path = QString::fromStdString(m_game_info->m_games[item->row()].snd0_path);
```
Comment (roamic) [inline src/qt_gui/game_list_frame.cpp]:
```suggestion
    const QString snd0path = QString::fromStdString(m_game_info->m_games[item->row()].snd0_path);
```

**Comment by georgemoralis** [comment]
@tGecko can you rebase this pr ? it is ok for merging

**Comment by tGecko** [comment]
will do later today 👍🏻

**Comment by tGecko** [comment]
@georgemoralis done.
I would like to implement navigating through the games list using the arrow keys as well, but I'll do that seperately.
(navigating works but it doesn't apply background image or play the soundtrack)

**Pull request merged**